proposals/solid-notifications.md

```diff
@@ -0,0 +1,270 @@
+
+In addition, the following terms are defined:
+
+**Notification Gateway API** -- an HTTP endpoint at which a client can negotiate an acceptable notification subscription location
+
+**Notification Subscription API** -- an HTTP endpoint at which a client can initiate a subscription to notification events for a particular set of resources
+
+**Solid Server Metadata Resource** - an RDF document that includes metadata about the Solid server
+
+## High-level Flow
+
+The following diagram shows the high level interactions involved in this flow. How a client retrieves an access token for step 5 is outside the scope of this document.
+
+![High level flow](solid-notifications-flow.png)
+
+## Discovery
+
+For any resource in a data pod, a client can discover the associated notification gateway API by fetching the Solid Server Metadata Resource. This resource can be retrieved by appending /.well-known/solid to the base URL of the data pod.
+
+```http
+GET /.well-known/solid
+```
+
+The server `MUST` be capable of serializing this resource as Turtle or JSON-LD.
+
+A sample representation of this resource might include the following:
+
+```http
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "notification_endpoint": "https://gateway.example",
+    ...
+}
+```
+
+**Note**: the notification panel should define a particular JSON-LD @context for use with JSON-LD serializations.
+
+**Note**: the use of a well-known resource is based on suggestions to avoid too many link headers, but that is not a settled issue.
+
+## Protocol
+
+The notification protocol involves two endpoints: a notification gateway API and a notification subscription API. A server `MUST` support JSON-LD in these interactions.
+
+### Notification Gateway
+Once the location of a notification gateway is discovered, an application will negotiate for a particular notification channel, based on acceptable features and protocols.
+
+The `type` array advertises which subscription channels a client is able to understand. The `features` array advertises which notification features are required by the client. The server selects one of the proposed subscription types and responds with a JSON-LD document that includes the selected protocol, the endpoint for negotiating a subscription and the supported features offered by that protocol.
+
+Authentication is not required at this endpoint.
+
+A sample interaction is described below. The type and feature names are included as examples.
+
+#### Request
+
+```http
+POST /gateway
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "type": ["WebSocketSubscription2021", "WebHookSubscription2021"],
+    "features": ["state", "rate"]
+}
+```
+
+#### Response
+
+```http
+HTTP/2
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "type": "WebSocketSubscription2021",
```

Comment:
For clarification, I assume the server will only respond with a(?) subscription channel that is supported by the client. What happens if there are none? Or if there is multiple? (In the latter case, I assume it will then be up to the client to decide which one to use).

Reply:
A server responds with either a single subscription channel or returns a 4xx error.

Reply:
#9